testgres/pubsub.py

```diff
@@ -0,0 +1,161 @@
+                in PostgreSQL documentation for more information)
+        """
+        self.name = subname
+        self.node = node
+        self.pub = publication
+
+        # connection info
+        conninfo = (
+            u"dbname={} user={} host={} port={}"
+        ).format(self.pub.dbname,
+                 self.pub.username,
+                 self.pub.node.host,
+                 self.pub.node.port)
+
+        query = (
+            "create subscription {} connection '{}' publication {}"
+        ).format(subname, conninfo, self.pub.name)
+
+        # additional parameters
+        if kwargs:
+            params = ','.join('{}={}'.format(k, v) for k, v in iteritems(kwargs))
+            query += " with ({})".format(params)
+
+        node.safe_psql(query, dbname=dbname, username=username)
+
+    def disable(self, dbname=None, username=None):
+        """
+        Disables the running subscription.
+        """
+        query = "alter subscription {} disable"
+        self.node.safe_psql(query.format(self.name),
+                            dbname=None,
+                            username=None)
+
+    def enable(self, dbname=None, username=None):
+        """
+        Enables the previously disabled subscription.
+        """
+        query = "alter subscription {} enable"
+        self.node.safe_psql(query.format(self.name),
+                            dbname=None,
+                            username=None)
+
+    def refresh(self, copy_data=True, dbname=None, username=None):
+        """
+        Disables the running subscription.
+        """
+        query = "alter subscription {} refresh publication with (copy_data={})"
+        self.node.safe_psql(query.format(self.name, copy_data),
+                            dbname=dbname,
+                            username=username)
+
+    def close(self, dbname=None, username=None):
+        """
+        Drops subscription
+        """
+        self.node.safe_psql("drop subscription {}".format(self.name),
+                            dbname=dbname, username=username)
+
+    def catchup(self, username=None):
+        """
+        Wait until subscription catches up with publication.
+
+        Args:
+            username: remote node's user name
+        """
+        if pg_version_ge('10'):
+            query = (
+                "select pg_current_wal_lsn() - replay_lsn = 0 "
+                "from pg_stat_replication where application_name = '{}'"
+            ).format(self.name)
+        else:
+            query = (
+                "select pg_current_xlog_location() - replay_location = 0 "
+                "from pg_stat_replication where application_name = '{}'"
+            ).format(self.name)
+
+        try:
+            # wait until this LSN reaches subscriber
+            self.pub.node.poll_query_until(
+                query=query,
+                dbname=self.pub.dbname,
+                username=username or self.pub.username,
+                max_attempts=60,
```

Comment:
IMHO it's better to replace `max_attempts=60` and `zero_rows_is_ok=True` with kwargs.